test/integration/scheduler_perf/util.go

```diff
@@ -33,8 +35,28 @@ import (
 	_ "k8s.io/kubernetes/plugin/pkg/scheduler/algorithmprovider"
 	"k8s.io/kubernetes/plugin/pkg/scheduler/factory"
 	"k8s.io/kubernetes/test/integration/framework"
+	"strconv"
 )
 
+var running_metrics_server bool
+
+// setupMetricsServer starts a metrics server so we can monitor scheduling latency/throughput/etc
+func setupMetricsServer() {
+	if !running_server {
+		running_server = true
+		glog.V(2).Info("starting metrics server")
+		mux := http.NewServeMux()
+		mux.Handle("/metrics", prometheus.Handler())
+		server := &http.Server{
+			Addr:    net.JoinHostPort("127.0.0.1", strconv.Itoa(int(9091))),
+			Handler: mux,
+		}
+		glog.Fatal(server.ListenAndServe())
+	} else {
```

Comment:
BTW, Not really sure why this function gets called more then once in the tests.  if we do decide to merge this then i can clean up this extra complexity by looking at the right place in the startup path to init the metrics server.